package.json

```diff
@@ -22,20 +22,20 @@
   "bin": "./bin/node-gyp.js",
   "main": "./lib/node-gyp.js",
   "dependencies": {
+    "env-paths": "^1.0.0",
     "glob": "^7.0.3",
     "graceful-fs": "^4.1.2",
     "mkdirp": "^0.5.0",
     "nopt": "2 || 3",
     "npmlog": "0 || 1 || 2 || 3 || 4",
-    "osenv": "0",
     "request": "^2.87.0",
     "rimraf": "2",
     "semver": "~5.3.0",
     "tar": "^3.1.3",
     "which": "1"
   },
   "engines": {
-    "node": ">= 4.0.0"
+    "node": ">= 6.0.0"
```

Comment:
This is tricky. We need more buy-in from other contributors.
@nodejs/node-gyp

Reply:
So far there have been no objections to dropping support for Node.js <6 for node-gyp v4 (https://github.com/nodejs/node-gyp/issues/1519). I also suggested dropping support for Node.js 6 which will be EOL in April.